modal/_container_io_manager.py

```diff
@@ -601,6 +602,22 @@ async def memory_restore(self) -> None:
         self.current_input_id = None
         self.current_input_started_at = None
 
+        # Patch torch to ensure it doesn't return CUDA unavailibility due to
+        # cached queries that executed during snapshot process. ref: MOD-3257
+        #
+        # perf: scanning sys.modules keys before import to avoid slow PYTHONPATH scanning.
+        if "torch" in sys.modules:
+            try:
+                import torch
+
+                torch.cuda.device_count = torch.cuda._device_count_nvml
```

Comment:
I doubt it would actually be much faster, but you probably could skip the `import` statement and just do

```
sys.modules["torch"].cuda.device_count = ...
```

And actually maybe checking for `"torch.cuda"` in sys.modules is an even narrower hack and less likely to get tripped up if people have a local `torch.py` or something in their modal root (i mean, I hope they don't, but you never know...)

Reply:
> if people have a local torch.py or something in their modal root 

those people should be called out, scolded by the logline! haha